### PR TITLE
st2 [2663][IMP] sale_order_secondary_unit_price account_invoice_secondary_unit_price

### PR DESCRIPTION
**Pull request by kanda999**
[2663](https://www.quartile.co/web?#id=2663&action=771&model=project.task&view_type=form&menu_id=505) 

add constrain if no seccondary uom entered but unit price is entered

### CODE REVIEW

File: account_invoice_secondary_unit_price/models/account_invoice_line.py [hunk 1]
```diff
@@ -37,3 +38,9 @@ def _onchange_uom_id(self):
         """
         super(AccountInvoiceLine, self)._onchange_uom_id()
         self.onchange_secondary_price()
+
+    @api.constrains("secondary_uom_id", "secondary_uom_price")
+    def _check_secondary_uom_id(self):
+        for line in self:
+            if (line.secondary_uom_price > 0) and (line.secondary_uom_id.id is False):
```
Comment (yostashiro) [inline account_invoice_secondary_unit_price/models/account_invoice_line.py]:
```suggestion
            if not line.secondary_uom_id and line.secondary_uom_price != 0.0:
```
@kanda999 I think this should work.  Can you check it.

File: account_invoice_secondary_unit_price/models/account_invoice_line.py [hunk 2]
```diff
@@ -37,3 +38,9 @@ def _onchange_uom_id(self):
...
+                raise ValidationError(_("Please enter Secondary uom."))
```
Comment (yostashiro) [inline account_invoice_secondary_unit_price/models/account_invoice_line.py]:
```suggestion
                raise ValidationError(_("Please enter secondary UoM or remove the secondary UoM price."))
```

File: account_invoice_secondary_unit/models/account_invoice_line.py [hunk 3]
```diff
@@ -86,3 +87,9 @@ def _onchange_product_id(self):
             self.secondary_uom_qty = 1.0
             self.onchange_secondary_uom()
         return res
+
+    @api.constrains("secondary_uom_id","secondary_uom_qty")
+    def _check_secondary_uom_id(self):
+        for line in self:
+            if line.secondary_uom_qty and line.secondary_uom_id == False:
```
Comment (AungKoKoLin1997) [inline account_invoice_secondary_unit/models/account_invoice_line.py]: if line.secondary_uom_qty > 0 and line.secondary_uom_id == False:

**Review by AungKoKoLin1997 — CHANGES_REQUESTED** [review]

File: account_invoice_secondary_unit/models/account_invoice_line.py (hunk 3 above)
Comment (kanda999) [reply account_invoice_secondary_unit/models/account_invoice_line.py]: Sorry , this line is not in the current commit.
Please watch file changes in the latest state.

File: account_invoice_secondary_unit_price/models/account_invoice_line.py (hunk 1 above)
Comment (kanda999) [reply account_invoice_secondary_unit_price/models/account_invoice_line.py]: It can work.

**Review by yostashiro — APPROVED** [review]
LGTM 👍

**Pull request merged**